src/app/ui/TaskBoard.tsx

```diff
@@ -59,28 +60,63 @@ export const TaskBoard = ({
   return (
     <AppMargin size={SizeofAppMargin.LARGE} py="18.5px">
       <Stack
-        direction="row"
         columnGap={2}
         sx={{
           overflowX: 'auto',
+          flexDirection: view === View.BOARD_VIEW ? 'row' : 'column',
         }}
       >
         {workflowStates.map((list, index) => {
-          return (
-            <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
-              <TaskColumn key={list.id} columnName={list.name} taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}>
-                {filterTaskWithWorkflowStateId(list.id).map((task, index) => {
-                  return (
-                    <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={task.id || ''}>
-                      <Box onClick={() => router.push(`/detail/${task.id}/iu?token=${token}`)} key={task.id} m="6px 0px">
-                        <TaskCard task={task} key={task.id} />
-                      </Box>
-                    </DragDropHandler>
-                  )
-                })}
-              </TaskColumn>
-            </DragDropHandler>
-          )
+          if (view === View.BOARD_VIEW) {
+            return (
+              <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
+                <TaskColumn
+                  key={list.id}
+                  columnName={list.name}
+                  taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}
+                >
+                  {filterTaskWithWorkflowStateId(list.id).map((task, index) => {
+                    return (
+                      <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={task.id || ''}>
+                        <Box onClick={() => router.push(`/detail/${task.id}/iu?token=${token}`)} key={task.id} m="6px 0px">
+                          <TaskCard task={task} key={task.id} />
+                        </Box>
+                      </DragDropHandler>
+                    )
+                  })}
+                </TaskColumn>
+              </DragDropHandler>
+            )
+          }
+          if (view === View.LIST_VIEW) {
+            return (
+              <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
+                <TaskRow
+                  key={list.id}
+                  columnName={list.name}
+                  taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}
```

Comment:
Isn't this function name a bit too long :stuck_out_tongue: 

Maybe something like `taskCountForWorkflowStateId` would be shorter

Reply:
Updated in c2e4e869a82b95c44e00ca9f7d8441ad4edc270d commit